src/Microsoft.AspNet.Mvc.DataAnnotations/DataTypeAttributeAdapter.cs

```diff
@@ -39,5 +39,10 @@ public DataTypeAttributeAdapter(DataTypeAttribute attribute, string ruleName, IS
             var errorMessage = GetErrorMessage(context.ModelMetadata);
             return new[] { new ModelClientValidationRule(RuleName, errorMessage) };
         }
+
+        public string GetErrorMessage(ModelMetadata metadata)
```

Comment:
Since this class and its subclasses are no longer just mapping to a "single client side validation rule", need to update all the doc comments.